rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.hpp

```diff
@@ -994,13 +997,28 @@ bool TypeSupport<MembersType>::serialize(
   assert(data);
   assert(payload);
 
-  auto ser = static_cast<eprosima::fastcdr::Cdr *>(data);
-  if (payload->max_size >= ser->getSerializedDataLength()) {
-    payload->length = static_cast<uint32_t>(ser->getSerializedDataLength());
-    payload->encapsulation = ser->endianness() ==
-      eprosima::fastcdr::Cdr::BIG_ENDIANNESS ? CDR_BE : CDR_LE;
-    memcpy(payload->data, ser->getBufferPointer(), ser->getSerializedDataLength());
-    return true;
+  auto ser_data = static_cast<SerializedData *>(data);
```

Comment:
The function could be strongly typed then, right? So this cast could go.

Reply:
Not possible, as it is the implementation of Fast-RTPS TopicDataType interface, declared in [this line](https://github.com/eProsima/Fast-RTPS/blob/fb91928914f7f8e47d0ef10bae26b207524b75ad/include/fastrtps/TopicDataType.h#L52)